### ISSUE TITLE
Enhancement: Cast ObjectId to string inside of MongooseMap.

### ISSUE THREAD

**Issue by HunterKohler** [Automattic/mongoose#9936]
## Enchancment
I am working on a project that uses the `MongooseMap` type and thought one simple line would help within the `get`, `set`, and `delete` _instance methods_. We just add a short conversion onto the following functions:
https://github.com/Automattic/mongoose/blob/aef25e09d54d272f14c93cbfa01d0986105e41ef/lib/types/map.js#L43-L44
https://github.com/Automattic/mongoose/blob/aef25e09d54d272f14c93cbfa01d0986105e41ef/lib/types/map.js#L51-L52
https://github.com/Automattic/mongoose/blob/aef25e09d54d272f14c93cbfa01d0986105e41ef/lib/types/map.js#L109-L110
Namely, insert: 
```javascript
if(key instanceof ObjectId) {
    key = key.toString();
}
```

Also require ObjectId in the header.
https://github.com/Automattic/mongoose/blob/aef25e09d54d272f14c93cbfa01d0986105e41ef/lib/types/map.js#L2-L4
```javascript
const ObjectId = require('./objectid')
```

## Example
This implicit cast would be more or less expected as ObjectId is so frequently represented as a string. This example just shows a case where data would be stored in conjunction with a list of references. We want to get the data upon user requests, and calling the user id on the Map's hash-table will obviously be faster than a linear scan (or at least less painfull to look at then a `toString` call every 10 seconds).

```javascript
const UserSchema = {
    name: String
}

const User = mongoose.model('User', UserSchema);

const GroupSchema = {
    members: {
        type: Map,
        of: { 
            read_receipts: Bool,
            share_location: Bool,
            follow_globals: Bool,
            nickname: String,
            role: {
                type: String
                enum: ['member', 'creator', 'admin']
            },
         }
     }
}

GroupSchema.methods.add
    = function add(userId, role = 'member') {
        this.members.set(userId, { role });
    };

GroupSchema.methods.remove
    = function remove(userId) {
        this.members.delete(userId);
     };

const Group = model('Group', GroupSchema);
```
## TL;DR
Add ObjectId conversion to map so we don't have to think about converting to string. Maintain object relationships with ObjectIds meant to represent other docs.

**Comment by vkarpov15** [comment]
Fixed by #9938
